core/camel-core-processor/src/main/java/org/apache/camel/processor/OnCompletionProcessor.java

```diff
@@ -190,8 +190,8 @@ protected static void doProcess(Processor processor, Exchange exchange) {
         } finally {
             // restore the options
             exchange.setRouteStop(stop);
-            if (failureHandled != null) {
-                exchange.setProperty(ExchangePropertyKey.FAILURE_HANDLED, failureHandled);
+            if (failureHandled) {
```

Comment:
I wonder if we have anywhere in the code, where we call: setFailureHandled(false), as then this code will not restore that value.

However I suspect we do not have that, as we use Boolean.TRUE before, and "remove" or "null" to say false/none. But may be worth to grep the source to find what we do

Reply:
It seems we don't do that. Neither IntelliJ nor `egrep -inR "setFailureHandled" *` found any instance of us setting the flag to false. I believe we should be fine with this one.